File: PythonSolutions/Neetcode150/array_problems/217_contains-duplicate.py
from typing import List

def containsDuplicate(nums: List[int]) -> bool:
    mySet = set()

    for n in nums:
        if n in mySet:
            return True
        
        mySet.add(n)
    
    return False

print(containsDuplicate([1,1,1,3,3,4,3,2,4,2]))

File: TypescriptSolutions/Neetcode150/array_problems/217_contains-duplicate.ts
function containsDuplicate(nums: number[]): boolean {
    const numsSet = new Set<number>();

    for (let i = 0; i < nums.length; i++) {
        if (numsSet.has(nums[i])) {
            return true;
        }

        numsSet.add(nums[i]);
    }

    return false;
};